src/libraries/System.Private.CoreLib/src/System/ISpanFormattable.cs

```diff
@@ -3,8 +3,19 @@
 
 namespace System
 {
-    internal interface ISpanFormattable
+    /// <summary>Provides functionality to format the string representation of an object into a span.</summary>
+    public interface ISpanFormattable : IFormattable
```

Comment:
> If this is now : IFormattable, we can remove the redundant constraint from methods like StringBuilder.AppendSpanFormattable

That's an internal method and an as such an implementation detail.  I expect we'll want to revisit integration in StringBuilder in general with this being public.  With the method as it currently stands, the constraint is there to avoid needing a cast, though we only ever call it today with value types, so the cast would evaporate.  I can clean that up as part of this.

> Additionally, are there any other types we'd want to add ISpanFormattable to?

This PR adds it to BigInteger.  I'd intended to add it to IPAddress, which already has a TryFormat method, but apparently didn't commit that; I'll fix that. We might also want to add it to some of the primitive drawing types, like Point.  But beyond that, I looked through the rest of our IFormattable implementations, and didn't see much else that looked interesting.  We could also look at ToString overrides for inspiration, but many of those just return an existing string; we'd only benefit in cases where ToString was doing something more, like concatenating two strings, and where we'd expect them to be used in frequent interpolations, rather than just for debugging.